plugins/beacon/ios/Tests/BaseBeaconPluginTests.swift

```diff
@@ -136,4 +136,105 @@ class BaseBeaconPluginTests: XCTestCase {
         ))
         wait(for: [expectation], timeout: 2)
     }
+    
+    func testCancelSpecificBeaconsUsingHooks() {
+        let context = JSContext()!
+        JSUtilities.polyfill(context)
+        
+        let cancelBeaconHandler = expectation(description: "Cancel Beacon Handler called")
+        let handlerExpectation = expectation(description: "Handler called")
+        handlerExpectation.isInverted = true
+        
+        let handler = MockHandler()
+        
+        let plugin = BaseBeaconPlugin<DefaultBeacon>(plugins: []) { (beacon) in
+            if handler.calls.isEmpty {
+                handler.handle(beacon.viewId!, beacon.data)
+                handlerExpectation.fulfill()
+            }
+        }
+        
+        plugin.context = context
+        plugin.setup(context: context)
+        
+        plugin.beacon(assetBeacon: AssetBeacon(
+            action: BeaconAction.clicked.rawValue,
+            element: BeaconElement.button.rawValue,
+            asset: BeaconableAsset(id: "test"),
+            data: .string(data: "example")
+        ))
+        
+        guard let hooks = plugin.hooks else {
+            XCTFail("Hooks are not initialized")
+            return
+        }
+        
+        hooks.cancelBeacon.tap { (arg1: JSValue, arg2: JSValue) -> Bool in
+            cancelBeaconHandler.fulfill()
+            if let action = arg1.toDictionary()?["action"] as? String, action == BeaconAction.clicked.rawValue {
+                return true
+            }
+            return false
+        }
+        
+        wait(for: [handlerExpectation, cancelBeaconHandler], timeout: 10)
+    }
+    
+    class MockHandler {
+        var calls: [(String, Any?)] = []
+        
+        func handle(_ action: String, _ data: Any? = nil) {
```

Comment:
i dont think this MockHandler is needed if we can test with expectations

Reply:
I'm using MockHandler to test this piece of code https://github.com/player-ui/player/pull/536/files#diff-f6cdb5dd4b4782a5f8b61d122af9dc5c90dad5cdbe669551df7e4e9ce855b37fR201